src/main/java/mekanism/common/item/interfaces/IJetpackItem.java

```diff
@@ -0,0 +1,163 @@
+import mekanism.common.util.MekanismUtils;
+import mekanism.common.util.MekanismUtils.ResourceType;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.phys.Vec3;
+
+public interface IJetpackItem {
+
+    boolean canUseJetpack(ItemStack stack);
+
+    JetpackMode getJetpackMode(ItemStack stack);
+
+    void useJetpackFuel(ItemStack stack);
+
+    default boolean isActive(ItemStack stack) {
+        return canUseJetpack(stack);
+    }
+
+    enum JetpackMode implements IIncrementalEnum<JetpackMode>, IHasTextComponent {
+        NORMAL(MekanismLang.JETPACK_NORMAL, EnumColor.DARK_GREEN, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_normal.png")),
+        HOVER(MekanismLang.JETPACK_HOVER, EnumColor.DARK_AQUA, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_hover.png")),
+        DISABLED(MekanismLang.JETPACK_DISABLED, EnumColor.DARK_RED, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_off.png"));
+
+        private static final JetpackMode[] MODES = values();
+        private final ILangEntry langEntry;
+        private final EnumColor color;
+        private final ResourceLocation hudIcon;
+
+        JetpackMode(ILangEntry langEntry, EnumColor color, ResourceLocation hudIcon) {
+            this.langEntry = langEntry;
+            this.color = color;
+            this.hudIcon = hudIcon;
+        }
+
+        @Override
+        public Component getTextComponent() {
+            return langEntry.translateColored(color);
+        }
+
+        @Nonnull
+        @Override
+        public JetpackMode byIndex(int index) {
+            return byIndexStatic(index);
+        }
+
+        public ResourceLocation getHUDIcon() {
+            return hudIcon;
+        }
+
+        public static JetpackMode byIndexStatic(int index) {
+            return MathUtils.getByIndexMod(MODES, index);
+        }
+    }
+
+    /**
+     * Gets the first found active jetpack from an entity, if one is worn.
+     * <br>
+     * If Curios is loaded, the curio slots will be checked as well.
+     *
+     * @param entity the entity on which to look for the jetpack
+     *
+     * @return the jetpack stack if present, otherwise an empty stack
+     */
+    @Nonnull
+    static ItemStack getActiveJetpack(LivingEntity entity) {
+        ItemStack chest = entity.getItemBySlot(EquipmentSlot.CHEST);
+        if (chest.getItem() instanceof IJetpackItem jetpackItem && jetpackItem.isActive(chest)) {
+            return chest;
+        } else if (Mekanism.hooks.CuriosLoaded) {
+            return CuriosIntegration.findFirstCurio(entity, s -> s.getItem() instanceof IJetpackItem jetpackItem && jetpackItem.isActive(s));
+        }
+        return ItemStack.EMPTY;
+    }
```

Comment:
I feel like this method and the one below could have a private helper that they use like this:
```java
private static ItemStack getJetpack(LivingEntity entity, Predicate<ItemStack> matcher) {
    ItemStack chest = entity.getItemBySlot(EquipmentSlot.CHEST);
    if (matcher.test(chest)) {
        return chest;
    } else if (Mekanism.hooks.CuriosLoaded) {
        return CuriosIntegration.findFirstCurio(entity, matcher);
    }
    return ItemStack.EMPTY;
}
```